server/selection_algorithm.go

```diff
@@ -0,0 +1,104 @@
+package server
+
+import (
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/golang/glog"
+	"math"
+	"math/rand"
+)
+
+type ProbabilitySelectionAlgorithm struct {
+	MinPerfScore float64
+
+	StakeWeight float64
+	PriceWeight float64
+	RandWeight  float64
+
+	PriceExpFactor float64
+}
+
+func (sa ProbabilitySelectionAlgorithm) Select(addrs []ethcommon.Address, stakes map[ethcommon.Address]int64, prices map[ethcommon.Address]float64, perfScores map[ethcommon.Address]float64) ethcommon.Address {
```

Comment:
🙌  Really nice that you were able to split this out - it should make it a lot easier for people to tweak it in the future without understanding the entirety of go-livepeer

Reply:
Yeah, that's the plan. Or we can even decide in the future to have another implementation of `SelectionAlgorithm` which will use webook (aka. plugin approach).